src/system/api/rqidApi.ts

```diff
@@ -114,18 +116,65 @@ export class Rqid {
       result.splice(0, 0, ...worldDocuments);
     }
 
+    const distinctWorldRqids: string[] = [
+      ...new Set(result.map((d) => d.data?.flags?.rqg?.documentRqidFlags?.id)),
+    ];
+
     if (["match", "all", "compendiums"].includes(scope)) {
-      const compendiaDocuments = await Rqid.documentsFromCompendia(
-        rqidRegex,
-        rqidDocumentType,
-        lang
-      );
+      let compendiaDocuments = await Rqid.documentsFromCompendia(rqidRegex, rqidDocumentType, lang);
+
       result.splice(result.length, 0, ...compendiaDocuments);
     }
 
     return result;
   }
 
+  /**
+   * Gets only the highest priority documents for each rqid that matches the Regex and
+   * language, with the highest priority documents in the World taking precedence over
+   * any documents
+   * in compendium packs.
+   * @param rqidRegex regex used on the rqid
+   * @param rqidDocumentType the first part of the wanted rqid, for example "i", "a", "je"
+   * @param lang the language to match against ("en", "es", ...)
+   * **match** same logic as fromRqid function,
+   * **all**: find in both world & compendia,
+   * **world**: only search in world,
+   * **compendiums**: only search in compendiums
+   */
+  public static async fromRqidRegexBest(
+    rqidRegex: RegExp | undefined,
+    rqidDocumentType: string, // like "i", "a", "je"
+    lang: string = "en"
+    // onlyFindBest: boolean = false
+  ): Promise<Document<any, any>[]> {
+    const allDocuments = await this.fromRqidRegexAll(rqidRegex, rqidDocumentType, lang, "all");
```

Comment:
I think you want scope "match" if you want it to behave the same way as `fromRqid`. "all" will give you the highest prio in the world **and** compendiums (if a doc in a compendium has the highest prio, you will return that instead of any world docs).

Reply:
Changing it to "match" definitely breaks it.  Let's say I have "a.fake-template" twice in the world, one with a higher prio, and "a.human-template" in the compendium.  With "match" I only get the best "a.fake-template".  With "all" I get both the best "a.fake-template" and "a.human-template" from the compendium.